core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -169,8 +172,34 @@ public boolean isOnlyIfSuccessful() {
         return onlyIfSuccessful;
     }
 
+    public boolean isOnSuccessful() {
+        return onSuccessful;
+    }
+
+    public boolean isOnFailed() {
+        return onFailed;
+    }
+
+    public boolean isOnAborted() {
+        return onAborted;
+    }
+
     @DataBoundSetter public final void setOnlyIfSuccessful(boolean onlyIfSuccessful) {
-        this.onlyIfSuccessful = onlyIfSuccessful;
+        this.onSuccessful = onlyIfSuccessful;
+        this.onFailed = false;
```

Comment:
This actually changes the behavior. When it was *disabled*, it archived *always*.

Why isn't this `@Deprecated` to begin with?